pkg/services/ngalert/store/instance_database.go

```diff
@@ -142,3 +143,17 @@ func (st DBstore) FetchOrgIds() ([]int64, error) {
 
 	return orgIds, err
 }
+
+func (st DBstore) DeleteAlertInstance(orgID int64, ruleUID string, labels models.InstanceLabels) error {
+	return st.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		labelTupleJSON, labelsHash, err := labels.StringAndHash()
+		if err != nil {
+			return err
+		}
+		_, err = sess.Exec("DELETE FROM alert_instance WHERE rule_org_id = ? AND rule_uid = ? AND labels = ? AND labels_hash = ?", orgID, ruleUID, labelTupleJSON, labelsHash)
```

Comment:
Do we need also the `labels` in the where condition given the `labels_hash` is derived from them and the `labels` column [is not included in the table primary key or an index](https://github.com/grafana/grafana/blob/8a3d7b64469ca4f260b3a712592bfc2a18397876/pkg/services/sqlstore/migrations/ualert/tables.go#L106).

Reply:
You're totally right, thank you!